packages/skill-lesson/lib/module-progress.ts

```diff
@@ -16,16 +14,25 @@ export async function getModuleProgress({
   moduleSlug: string
   userId: string
 }) {
-  const module = await getModule(moduleSlug)
+  const module = await getModuleStructure(moduleSlug)
 
-  const allModuleLessons =
-    module.sections.length > 0
-      ? module.sections
-          .filter((section: Section) => section?.lessons?.length)
-          .flatMap((section: Section) => section.lessons)
-      : module.lessons
+  type LessonStructure = NonNullable<ResourceStructure['resources']>[0]
 
-  const lessonIds = allModuleLessons.map((lesson: Lesson) => lesson._id)
+  const allModuleSections: Array<ResourceStructure> = []
+  const allModuleLessons: Array<LessonStructure> = []
```

Comment:
create a list of all lessons and all sections for the current module which get used throughout the rest of the function to compute completion stats